src/modules/launcher/Plugins/Microsoft.Plugin.Folder/Sources/QueryInternalDirectory.cs

```diff
@@ -0,0 +1,193 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using Microsoft.Plugin.Folder.Sources.Result;
+using Wox.Plugin;
+
+namespace Microsoft.Plugin.Folder.Sources
+{
+    public class QueryInternalDirectory : IQueryInternalDirectory
+    {
+        private readonly FolderSettings _settings;
+        private readonly IQueryFileSystemInfo _queryFileSystemInfo;
+
+        private static readonly HashSet<char> SpecialSearchChars = new HashSet<char>
+        {
+            '?', '*', '>',
+        };
+
+        private static string _warningIconPath;
+
+        public QueryInternalDirectory(FolderSettings folderSettings, IQueryFileSystemInfo queryFileSystemInfo)
+        {
+            _settings = folderSettings;
+            _queryFileSystemInfo = queryFileSystemInfo;
+        }
+
+        private static bool HasSpecialChars(string search)
+        {
+            return search.Any(c => SpecialSearchChars.Contains(c));
+        }
+
+        public static SearchOption GetSearchOptions(string actionKeyword)
+        {
+            // give the ability to search all folder when starting with >
+            if (actionKeyword?.StartsWith(">", StringComparison.InvariantCulture) == true)
```

Comment:
When I look up the wildcard '>' at [MSDN](https://docs.microsoft.com/en-us/dotnet/api/system.io.directoryinfo.enumeratefilesysteminfos?view=netcore-3.1) I can't find a special meaning for it. So it looks like this is custom behaviour that we are querying here. Now I know that '>' is a forbidden character for a file or directory, so we won't get any collision there, but how are we have to handle the following cases:

* C:\>>>>>>>
* C:\Te>st\
* C:\Test\>\test
* C:\Test\>\test\>*.jpg

For now I'm removing every instance of '>' in the query itself. But this could lead to weird behaviour I think. See my last case, probably the directory is pointing to somewhere else than people think.

Reply:
@P-Storm.  As you mentioned the '>' is forbidden by directories.   That is why I was proposing the `incompleteName` name be used for `GetSearchOptions`.   Prior to your change none of the test cases you provided would find anything, and I think that is to be expected as the  '>' character isn't valid for directories.    I do think we want to search all directories in the `C:\Test\test\>*.jpg` case though, which your latest change should fix, my only concern there would be that allowing the cases you have above would incur a perf hit despite semantically not making any sense. 

We inherited this behavior from the Wox plugins.  Feel free to start another discussion with @crutkas if you would like to propose a change to the syntax.